src/main/java/com/google/sps/servlets/schedule/comms/ReadUserSettingsFromDBServlet.java

```diff
@@ -0,0 +1,141 @@
+import java.util.List;
+import javax.servlet.RequestDispatcher;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.security.GeneralSecurityException;
+import java.util.Collections;
+import java.util.List;
+
+import org.json.simple.JSONObject;
+import org.json.simple.JSONArray;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+
+/** Servlet that gets Calendar settings. */
+/** ReadUserSettingsFromDBServlet simply reads from 
+ * To display calendar settings we need AT LEAST 2 pieces of information 
+ * 1) a calendar ID 2) timezone. This will be displayed using an iFrame.
+ * 
+ * As of now, we display the user's primary_idcalendar and the study schedule ID.
+ * 
+ * We check if we have a calendar already on saved on the Servlet session, then we just send that one
+ * Else we check if the user has a Study Schedule on their list of calendars. Then we send that one
+ * Else we return an error that there is no calendar to display.
+ */
+@WebServlet("/getSettingsState")
+public class ReadUserSettingsFromDBServlet extends HttpServlet {
+  private Datastore DB = new Datastore();
+
+  @Override
+  /** doGet
+   * 
+   * The fields are as follows: 
+   * 
+   * @param request 
+   * @return response JSON object consisting of fields required to display the object
+   */
+  public void doGet(HttpServletRequest request, HttpServletResponse response) 
+    throws IOException {
+
+    String accessToken = (String) request.getSession(false).getAttribute("access_token");
+
+    Boolean isValid = new HTTP().isAccessTokenValid(accessToken);
+    if (!isValid) {
+      response.sendRedirect("/home.html");
+      return;
+    }
+
+    // Check to make sure the user has a settings page ready to go to read from in the database!
+    GetCalendar getCalendar = new GetCalendar();
+    String json = new HTTP().get(getCalendar.createGetCalendarURL("primary", accessToken));
+    JSONObject jsonObject = new HTTP().parseJSON(json);
+    String primary_id = (String) jsonObject.get("id");
+    if (DB.getUser(primary_id) == null) {
+      response.sendRedirect("/home.html");
+      return;
+    }
+
+    try {
+      String defaults = request.getParameter("defaults");
+      Boolean getDefaults = Boolean.valueOf(defaults);
+      if (getDefaults) DB.applyDefaults(primary_id);
+    } catch (Exception e) {
+      System.out.println(e);
+    }
+
+    getSettingsFromDB(primary_id, response);
+  }
+
+  public void getSettingsFromDB(String primary_id, HttpServletResponse response) {
+    JSONObject sendJSON = new JSONObject();
+    try {
+      response.setContentType("application/json");
+      String description = DB.getUserSetting(primary_id, "description");
```

Comment:
In the future, a better way to go about this may be to have a list of setting params that we care about and then iterate through each of them instead of repeating similar code from line 102-125.